app/search/controller.ts

```diff
@@ -1,46 +1,96 @@
 import Store from '@ember-data/store';
+import { A } from '@ember/array';
 import Controller from '@ember/controller';
 import { action } from '@ember/object';
 import { inject as service } from '@ember/service';
 import { waitFor } from '@ember/test-waiters';
-import { task } from 'ember-concurrency';
-import { taskFor } from 'ember-concurrency-ts';
 import { tracked } from '@glimmer/tracking';
+import { task, timeout } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
 import Media from 'ember-responsive';
 
 import MetadataPropertySearchModel from 'ember-osf-web/models/metadata-property-search';
+export interface Filter {
+    property: string;
+    value: string;
+}
+
+const searchDebounceTime = 100;
 
 export default class SearchController extends Controller {
     @service store!: Store;
     @service media!: Media;
+    @service toast!: Toastr;
 
     queryParams = ['q', 'page', 'sort'];
-    @tracked q?: string;
-    @tracked page?: number;
-    @tracked sort?: string;
+    @tracked q?: string = '';
+    @tracked seachBoxText?: string = '';
+
+    @tracked page?: number = 1;
+    @tracked sort?: string = '-relevance';
+    @tracked activeFilters = A<Filter>([]);
 
     @tracked propertySearch?: MetadataPropertySearchModel;
 
-    get showSidenavToggle() {
+    get showSidePanelToggle() {
         return this.media.isMobile || this.media.isTablet;
     }
 
+    get filterableProperties() {
+        if (!this.propertySearch) {
+            return [];
+        }
+        return this.propertySearch.get('searchResultPage');
+    }
+
     @action
-    onKeyPress(event: KeyboardEvent) {
-        if (event.key === 'Enter') {
-            taskFor(this.doSearch).perform();
+    doSearch() {
+        this.q = this.seachBoxText;
+        taskFor(this.search).perform();
+    }
+
+    @action
+    toggleFilter(filter: Filter) {
+        const filterIndex = this.activeFilters.findIndex(
+            f => f.property === filter.property && f.value === filter.value,
+        );
+        if (filterIndex > -1) {
+            this.activeFilters.removeAt(filterIndex);
+        } else {
+            this.activeFilters.pushObject(filter);
+        }
+        this.doSearch();
+    }
+
+    @action
+    ingestQueryParams() {
+        const { q } = this;
+        if (q) {
+            this.seachBoxText = q;
         }
     }
 
-    @task
+    @task({ restartable: true, on: 'init' })
```

Comment:
Hmm. It's a shame that the page-load search is always going to wait the debounce time to load the initial results. It might be worth breaking out a separate `userSearch()` task that does the debounce and calls the `search()` task rather than having it all happen in one task.